src/Microsoft.Health.Fhir.CosmosDb/Features/Metrics/CosmosMetricProcessor.cs

```diff
@@ -0,0 +1,119 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using AngleSharp.Common;
+using EnsureThat;
+using MediatR;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Health.CosmosDb.Features.Storage;
+using Microsoft.Health.Fhir.Core.Features.Context;
+
+namespace Microsoft.Health.Fhir.CosmosDb.Features.Metrics
+{
+    public class CosmosMetricProcessor : ICosmosMetricProcessor
+    {
+        private readonly IFhirRequestContextAccessor _fhirRequestContextAccessor;
+        private readonly IMediator _mediator;
+        private readonly ILogger<CosmosMetricProcessor> _logger;
+
+        public CosmosMetricProcessor(IFhirRequestContextAccessor fhirRequestContextAccessor, IMediator mediator, ILogger<CosmosMetricProcessor> logger)
+        {
+            EnsureArg.IsNotNull(fhirRequestContextAccessor, nameof(fhirRequestContextAccessor));
+            EnsureArg.IsNotNull(mediator, nameof(mediator));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _fhirRequestContextAccessor = fhirRequestContextAccessor;
+            _mediator = mediator;
+            _logger = logger;
+        }
+
+        public void ProcessResponse<T>(T resourceResponseBase)
+            where T : IResourceResponseBase
+        {
+            ProcessResponse(resourceResponseBase.SessionToken, resourceResponseBase.RequestCharge, resourceResponseBase.CollectionSizeUsage, resourceResponseBase.StatusCode);
```

Comment:
We should check with Yazan but I think we don't want to publish the CollectionSizeUsage here. I've found that the number was unreliable and does not match the actual collection size (i think it was more like partition size or something like that).